configurations/morcore2esuites/Configuration_TaakApplicatieApi.xml

```diff
@@ -5,42 +5,65 @@
             <ApiListener
                 name="ListTaakTypes"
                 uriPattern="/taaktypes" />
-
         </Receiver>
 
         <Pipeline>
             <Exits>
                 <Exit name="Exit" state="SUCCESS" />
-            </Exits>
-
-            <EchoPipe name="Hello" getInputFromFixedValue="World!"></EchoPipe>
+                <Exit name="Reject" state="REJECTED"/>
+                <Exit name="failure" state="ERROR"/>
+             </Exits>
+            
+            <fixedResultPipe name="InputTaaktypesLijst" filename="responses/ZaakTypes.xml"/>
+
+            <XmlValidatorPipe name="ValidateInputMessage" schema="xsd/validateGetTaakTypes.xsd" root="root"/>
+                <Forward name="jsonpipe" path="XmlToJson"/>
+                <Forward name="failure" path="failure"/>
+            
+            <JsonPipe name="XmlToJson" direction="XML2JSON">
+                <Forward name="success" path="Exit"/>
+                <Forward name="failure" path="Reject"/>
+            </JsonPipe>
         </Pipeline>
     </Adapter>
 
     <Adapter name="Action_GetTaakType">
         <Receiver name="GetTaakType">
             <ApiListener
                 name="GetTaakType"
-                uriPattern="/taaktypes/{uuid}" />
-
+                uriPattern="/taaktypes/{uuid}"/>
         </Receiver>
 
         <Pipeline>
             <Exits>
                 <Exit name="Exit" state="SUCCESS" />
-            </Exits>
+                <Exit name="Reject" state="REJECTED"/>
+                <Exit name="failure" state="ERROR"/>
+             </Exits>
+
+            <fixedResultPipe name="InputTaakType" filename="responses/ZaakTypes.xml"/>
 
-            <EchoPipe name="Hello" getInputFromFixedValue="World!"></EchoPipe>
+            <XmlValidatorPipe name="ValidateInputMessage" schema="xsd/validateGetTaakTypes.xsd" root="root"/>
```

Comment:
A validator pipe here does not fulfil it's function. In this situation you want to use the validator as an output validator. That means it should be at or near the end of the pipeline so it can check the final output.